### PR TITLE
Document SIWE

### LINKED ISSUE #765: Add technical documentation for SIWE support
Documentation should cover:
* Thorough coverage of MetaMask's support for [EIP-4361](https://eips.ethereum.org/EIPS/eip-4361)
* MetaMask UI/UX behavior details that may be important for developers
* Include admonitions for current deviations from spec or gotchas

### PR DESCRIPTION
**Pull request by alexandratran**
Document how to use MetaMask's SIWE feature. Fixes #765.

Preview: https://docs.metamask.io/765-siwe/wallet/how-to/use-siwe/

### CODE REVIEW

File: wallet/how-to/use-siwe.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+# JavaScript
+
+```javascript
+const siweSign = async (siweMessage) => {
+  try {
+    const from = accounts[0];
+    const msg = `0x${Buffer.from(siweMessage, 'utf8').toString('hex')}`;
+    const sign = await ethereum.request({
+      method: 'personal_sign',
+      params: [msg, from, 'Example password'],
```
Comment (ziad-saab) [inline wallet/how-to/use-siwe.md:60]: Should we have some information/details about what the "Example password" is, and what it's used for? I myself have always been confused about it, and I don't see it explained anywhere.

Comment (ziad-saab) [reply wallet/how-to/use-siwe.md:60]: @alexandratran turns out that there's no "example password" parameter. can we remove it everywhere as part of this PR as well?

**Comment by ziad-saab** [comment]
@alexandratran can we take advantage of this PR to remove the other password here:

https://docs.metamask.io/wallet/how-to/sign-data/#example-1

**Review by vandan — APPROVED** [review]
lgtm

**Review by ziad-saab — APPROVED** [review]

**Review by httpJunkie — COMMENTED** [review]
Looks GTM, quick review at the airport.. But the grammar looks good, it's easy and short. Code samples look correct considering VanillaJS example.

**Pull request merged**